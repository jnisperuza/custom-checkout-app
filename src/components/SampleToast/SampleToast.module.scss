@import "../../styles/classes";

.sampleToast {
    @extend .PrimaryButton;
    width: 100px;

    &::after {
        content: "Toast";
    }
}