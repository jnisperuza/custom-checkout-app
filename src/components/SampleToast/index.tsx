import { useSnackbar } from 'notistack';

import styles from './SampleToast.module.scss';

/**
 * https://notistack.com/api-reference
 */
function SampleToast() {
    const { enqueueSnackbar } = useSnackbar();

    const handleClick = () => {
        enqueueSnackbar({
            message: 'This is a success alert',
            type: 'success'
        });
    }

    return (
        <button
            type='button'
            className={styles.sampleToast}
            onClick={handleClick} />
    )
}

export default SampleToast;