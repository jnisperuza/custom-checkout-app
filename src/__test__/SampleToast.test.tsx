import React from 'react';
import { render, screen } from '@testing-library/react';
import SampleToast from '../components/SampleToast';

const mockEnqueue = jest.fn();

jest.mock('notistack', () => ({
    ...jest.requireActual('notistack'),
    useSnackbar: () => {
        return {
            enqueueSnackbar: mockEnqueue
        };
    }
}));

describe('SampleToast tests', () => {
    it('should call SampleToast correctly', () => {
        const { container } = render(<SampleToast />);
        expect(container).toBeTruthy();
    });
    it('should have class sampleToast', () => {
        const { container } = render(<SampleToast />);
        const button = container.getElementsByClassName('sampleToast');
        expect(button.length).toBe(1);
    });
    it('renders correctly', () => {
        const { container } = render(<SampleToast />);
        expect(container).toMatchSnapshot();;
    });
});