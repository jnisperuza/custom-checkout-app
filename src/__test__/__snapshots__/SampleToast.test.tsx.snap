// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleToast tests renders correctly 1`] = `
<div>
  <button
    class="sampleToast"
    type="button"
  />
</div>
`;
